protected/humhub/modules/file/models/File.php

```diff
@@ -89,9 +109,17 @@ public function rules()
     {
         return [
             [['mime_type'], 'string', 'max' => 150],
-            [['mime_type'], 'match', 'not' => true, 'pattern' => '/[^a-zA-Z0-9\.ä\/\-\+]/', 'message' => Yii::t('FileModule.base', 'Invalid Mime-Type')],
+            [
+                ['mime_type'],
+                'match',
+                'not' => true,
+                'pattern' => '/[^a-zA-Z0-9\.ä\/\-\+]/',
+                'message' => Yii::t('FileModule.base', 'Invalid Mime-Type')
+            ],
             [['file_name', 'title'], 'string', 'max' => 255],
             [['size'], 'integer'],
+            [['state'], 'integer'], // ToDo: merge with size after all fields have been added
+            [['state'], 'unique', 'targetAttribute' => ['state', 'guid']],
```

Comment:
Should it be possible to have multiple entries in the file table with the same GUID but different states?

Reply:
I have not really looked at the code again, just divided it to the different fields. 😊

I guess this was not really thought through. The idea came from the possibility that a file might be in draft-state, before overwriting the published one. But then I guess later I solved that with having a draft variant instead...